cpp/src/arrow/flight/sql/odbc/CMakeLists.txt

```diff
@@ -21,6 +21,8 @@ add_subdirectory(flight_sql)
 add_subdirectory(odbcabstraction)
 add_subdirectory(tests)
 
+add_definitions(-DUNICODE=1)
```

Comment:
May want to revisit this and instead define SQL_NOUNICODEMAP to let you implement the ANSI functions directly. I have not tried building a driver this way.

Reply:
We should support unicode first, rather than ANSI first really. What I mean is that if you define SQL_NOUNICODEMAP, you get separate entry points for unicode and ANSI functions (ie SQLPrepare is ANSI and SQLPrepareW is unicode). This is the way I've supported unicode + ANSI in the past. For Windows this doesn't really matter.

Reply:
Since we are still supporting unicdoe first, I would prefer using `-DUNICODE=1` to define unicode functions. This is the way Apache Ignite ODBC defined unicode support so I will follow this way